esphome/components/ruuvi_ble/ruuvi_ble.cpp

```diff
@@ -0,0 +1,107 @@
+#include "ruuvi_ble.h"
+#include "esphome/core/log.h"
+
+#ifdef ARDUINO_ARCH_ESP32
+
+namespace esphome {
+namespace ruuvi_ble {
+
+static const char *TAG = "ruuvi_ble";
+
+bool parse_ruuvi_data_byte(uint8_t data_type, uint8_t data_length, const uint8_t *data, RuuviParseResult &result) {
+  switch (data_type) {
+    case 0x03: {  // RAWv1
+      if (data_length != 16)
+        return false;
+
+      const float humidity = float(data[0]) * 0.5f;
+
+      const uint8_t temperatureSign = uint8_t(data[1] >> 7 & 1);
+      const int8_t temperatureBase = int8_t(data[1] & 0x7F);
+      const float temperatureFraction = float(data[2]) / 100.0f;
+      const float temperature = float(temperatureBase) + temperatureFraction;
+
+      const uint8_t pressureHi = uint8_t(data[3] & 0xFF);
+      const uint8_t pressureLo = uint8_t(data[4] & 0xFF);
+      const float pressure = float((pressureHi * 256.0f + 50000.0f + pressureLo) / 100.0f);
```

Comment:
```suggestion
      const float pressure = (pressureHi * 256.0f + 50000.0f + pressureLo) / 100.0f;
```

Just an FYI, the final float() here is not rally necessary because the division by a float already promotes it to a float.